src/server/object_services/object_server.js

```diff
@@ -1572,28 +1573,20 @@ function throw_if_maintenance(req) {
 function check_quota(bucket) {
     if (!bucket.quota) return;
 
-    let used_percent = system_utils.get_bucket_quota_usage_percent(bucket, bucket.quota);
-
-    if (used_percent >= 100) {
-        const bucket_used = bucket.storage_stats && size_utils.json_to_bigint(bucket.storage_stats.objects_size);
-        const message = `the bucket ${bucket.name} used storage(${
-                size_utils.human_size(bucket_used)
-            }) exceeds the bucket quota (${
-                size_utils.human_size(bucket.quota.value)
-            })`;
-        Dispatcher.instance().alert('MAJOR',
-            system_store.data.systems[0]._id,
-            `Bucket ${bucket.name.unwrap()} exceeded its configured quota of ${
-                    size_utils.human_size(bucket.quota.value)
-                }, uploads to this bucket will be denied`,
-            Dispatcher.rules.once_daily);
+    let quota = new Quota(bucket.quota);
+    const alerts = [];
+    const major_messages = [];
+    Quota.add_quota_alerts(system_store.data.systems[0]._id, bucket, quota, alerts);
+    alerts.forEach(function(alert) {
```

Comment:
in general, prefer for-of instead of forEach, it's more readable in my opinion
```suggestion
    for (const alert of alerts) {
```

Reply:
Done